x86/vmx.c

```diff
@@ -1276,7 +1280,11 @@ static void init_vmcs_host(void)
 	vmcs_write(HOST_SEL_FS, KERNEL_DS);
 	vmcs_write(HOST_SEL_GS, KERNEL_DS);
 	vmcs_write(HOST_SEL_TR, TSS_MAIN);
+	#ifdef __x86_64__
```

Comment:
I think most of these inline idef's could be avoided if we instead did something like:

static void init_vmcs_host(void)
{
     #ifdef __x86_64__
     struct descriptor_table_ptr *tss_descr_ptr = tss_descr->base;
     #else
     struct descriptor_table_ptr *tss_descr_ptr = &tss_descr.base;
     #endif
     ...
}

This functions seems to be the only place that runs into this issue. If it wasn't, I'd probably suggest to squirrel away the logic I suggested above into a `get_tss_descr_ptr()` helper. However, I don't think it's necessary here.

What do you think?

Reply:
I just discovered that `x86/vmx.c` is only used in x86_64, not in i386, see:

https://github.com/marc-orr/KVM-Unit-Tests-dev-fork/blob/08e6e2cd4c17cfc62f62f5f7f2ba426b029d17c3/x86/Makefile.x86_64#L20

In this case we don't need to care about the i386 `tss_descr` type as it's never going to happen. So I can directly change the type.

Reply:
Ack. Makes sense.